src/main/java/biz/netcentric/maven/extension/repofromenv/FromEnvReposConfigurationProcessor.java

```diff
@@ -41,47 +43,80 @@
     static final String PROFILE_ID_REPOSITORIES_FROM_ENV = "repositoriesFromSysEnv";
     static final String REPO_ID_PREFIX = "sysEnvRepo";
 
+    static final String ENV_PROP_MVN_SETTINGS_REPO_VERBOSE = "MVN_SETTINGS_REPO_LOG_VERBOSE";
+
     @Inject
     private Logger logger;
 
+    private boolean isVerbose;
+    
     @Override
     public void process(CliRequest cliRequest) throws Exception {
-
-        List<RepoFromEnv> reposFromEnv = getReposFromEnv(System.getenv());
+        
+        Map<String, String> sysEnv = System.getenv();
+        isVerbose = Boolean.valueOf(sysEnv.get(ENV_PROP_MVN_SETTINGS_REPO_VERBOSE));
```

Comment:
Why is this needed? Is the verbosity level not picked up automatically from the -X parameter?

Reply:
It is, but if you run a typical build with `-X` you get a log file which is 50x larger than normal - allowing to set this also via system property allows to specifically get what you want (while the regular log "stays normal")

Reply:
OK I would still leverage the log api rather that the if/else - so `if (LOG.isDebugEnabled() || isVerbose)` .... - the output should still appear as debug, not info. Also, this sets the debug on an environmental variable level - wouldn't it make more sense to use a parameter to maven for one off 'debug' runs, or allow both?